src/lib/gadgets/bitwise.ts

```diff
@@ -11,7 +11,46 @@ import {
 } from './common.js';
 import { rangeCheck64 } from './range-check.js';
 
-export { xor, and, rotate };
+export { xor, not, and, rotate };
+
+function not(a: Field, length: number, checked: boolean = false) {
+  // check that input length is positive
+  assert(length > 0, `Input length needs to be positive values.`);
+
+  // Check that length does not exceed maximum field size in bits
+  assert(
+    length <= Field.sizeInBits(),
```

Comment:
this needs to be `length < Field.sizeInBits()` - see https://github.com/MinaProtocol/mina/blob/develop/src/lib/crypto/kimchi_backend/gadgets/bitwise.ml#L514

This is also the reason why the unit tests fail occasionally if the inpuits are greater than `2n ** 254n`

Reply:
Great catch @Trivo25! I was trying to figure out why the test occasionally failed at very large inputs.